library/src/scala/runtime/LazyVals.scala

```diff
@@ -119,11 +163,20 @@ object LazyVals {
 
 
   object Names {
+    final val controlState = "LazyValControlState"
```

Comment:
True. This is the approach that was used in previous implementations so I went ahead with it and just adjusted it, but I agree that it would make more sense to put it in the compiler.